frontend/components/usage.tsx

```diff
@@ -111,16 +117,19 @@ function StyleExamples({ baseUrl }) {
     </QueryParamTable>
   )
 }
-StyleExamples.propTypes = {
-  baseUrl: PropTypes.string.isRequired,
-}
 
 function NamedLogos() {
-  const renderLogo = logo => <LogoName key={logo}>{logo}</LogoName>
+  const renderLogo = (logo: string): JSX.Element => (
+    <LogoName key={logo}>{logo}</LogoName>
+  )
   const [first, ...rest] = shieldsLogos
-  return [renderLogo(first)].concat(
-    rest.reduce((result, logo) => result.concat([', ', renderLogo(logo)]), [])
+  const result = ([renderLogo(first)] as (JSX.Element | string)[]).concat(
+    rest.reduce(
+      (result, logo) => result.concat([', ', renderLogo(logo)]),
+      [] as (JSX.Element | string)[]
+    )
   )
+  return <>{result}</>
```

Comment:
I think I am starting to see what you mean with some of the tests we had for these components. Perhaps it's just the diff and/or the addition of the types, but seems like there's a bit of logic involved here that I personally wouldn't mind having a test for.

However, I also see that the existing spec file we had for this component wasn't testing this at all (at least AFAICT)

Reply:
Across all the components, the existing tests were checking that the code could run, and running a basic confidence check for a substring in the result. None of them were testing logic, per se.

To take this component as an example: there's a lot of stuff in it, though it's generating what is essentially static content. When this code changes we have to look at the page to test it, and I don't think it's going to spontaneously break unless it's something related to common styling, which tests aren't likely to catch. If this code were to break it would likely be type-related (e.g. a component changing which props it accepts) and the type checker will now catch that.

We do have components with a significant amount of logic. It might be useful to have tests for that logic, though I'm more interested in making the frontend more impressive and the behaviors better. I think the effort is better put into doing that than in writing tests, and I'm also not sure that the tests will help that process.

Reply:
Thanks for the context. Also just to clarify, my comment was not intended to be a suggestion (subtle or otherwise) that we needed a test here (or anywhere else) as part of this work to switch over to TypeScript. It was really just me thinking "out loud" 😄 

I still have not made the time to dig into this portion of the codebase to get familiar with it and find ways to contribute. When I am working with new code and/or refactoring, I like having an automated test to give me confidence that everything is functioning correctly and that I didn't break anything.

My thought here while looking at some of these changes was that if I had been the one working on them, that a test would have helped me personally in my dev loop, especially as someone not familiar with React/this part of our codebase.